### ISSUE TITLE
Tagging not working stable on cloud.nextcloud.com

### ISSUE THREAD

**Issue by mathiasconradt** [nextcloud/server#10774]
Tagging files does not always work on cloud.nextcloud.com, current version as of 21.08.2018

**Expected behaviour:** 

Tags show including the input field to add more tags

**Actual behaviour:** 

Nothing happens upon click, not even any JS error in the dev console.

This only happen sometimes, sometimes it works.

**How to reproduce it:**

Reload the browser page with Nextcloud file explorer in a folder that contains multiple files.
Open document details of one file, click on "Tags": the tag input box appears.
Now open the document details of another file (without reloading the browser!), click on "Tags": the tag input box does not appear.

**Comment by mathiasconradt** [comment]
@MorrisJobke Enabled dev console, even "disable cache" is activated. Still able to reproduce the error. I can show it to you via NC Talk Screenshare if you or others want, just ping me.

**Comment by blizzz** [comment]
@mathiasconradt cannot reproduce it either. What does the POST request to https://cloud.nextcloud.com/remote.php/dav/systemtags/ result in?

**Comment by mathiasconradt** [comment]
There is no POST request in the devl tools, only a PROPFIND


The response of the first PROPFIND returns proper values:

`PROPFIND https://cloud.nextcloud.com/remote.php/dav/systemtags/`

```
<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns"><d:response><d:href>/remote.php/dav/systemtags/</d:href><d:propstat><d:prop><oc:id/><oc:display-name/><oc:user-visible/><oc:user-assignable/><oc:can-assign/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/systemtags/6</d:href><d:propstat><d:prop><oc:id>6</oc:id><oc:display-name>Invoice</oc:display-name><oc:user-visible>true</oc:user-visible><oc:user-assignable>true</oc:user-assignable><oc:can-assign>true</oc:can-assign></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/systemtags/9</d:href><d:propstat><d:prop><oc:id>9</oc:id><oc:display-name>Marketing</oc:display-name><oc:user-visible>true</oc:user-visible><oc:user-assignable>true</oc:user-assignable><oc:can-assign>true</oc:can-assign></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/systemtags/7</d:href><d:propstat><d:prop><oc:id>7</oc:id><oc:display-name>RESTRICTED</oc:display-name><oc:user-visible>true</oc:user-visible><oc:user-assignable>true</oc:user-assignable><oc:can-assign>true</oc:can-assign></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/systemtags/8</d:href><d:propstat><d:prop><oc:id>8</oc:id><oc:display-name>fujitsu</oc:display-name><oc:user-visible>true</oc:user-visible><oc:user-assignable>true</oc:user-assignable><oc:can-assign>true</oc:can-assign></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/systemtags/3</d:href><d:propstat><d:prop><oc:id>3</oc:id><oc:display-name>pi</oc:display-name><oc:user-visible>true</oc:user-visible><oc:user-assignable>true</oc:user-assignable><oc:can-assign>true</oc:can-assign></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/systemtags/5</d:href><d:propstat><d:prop><oc:id>5</oc:id><oc:display-name>review</oc:display-name><oc:user-visible>true</oc:user-visible><oc:user-assignable>true</oc:user-assignable><oc:can-assign>true</oc:can-assign></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/systemtags/1</d:href><d:propstat><d:prop><oc:id>1</oc:id><oc:display-name>snap</oc:display-name><oc:user-visible>true</oc:user-visible><oc:user-assignable>true</oc:user-assignable><oc:can-assign>true</oc:can-assign></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/systemtags/10</d:href><d:propstat><d:prop><oc:id>10</oc:id><oc:display-name>test</oc:display-name><oc:user-visible>true</oc:user-visible><oc:user-assignable>true</oc:user-assignable><oc:can-assign>true</oc:can-assign></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/systemtags/11</d:href><d:propstat><d:prop><oc:id>11</oc:id><oc:display-name>test2</oc:display-name><oc:user-visible>true</oc:user-visible><oc:user-assignable>true</oc:user-assignable><oc:can-assign>true</oc:can-assign></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/systemtags/2</d:href><d:propstat><d:prop><oc:id>2</oc:id><oc:display-name>ubuntu core</oc:display-name><oc:user-visible>true</oc:user-visible><oc:user-assignable>true</oc:user-assignable><oc:can-assign>true</oc:can-assign></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/systemtags/4</d:href><d:propstat><d:prop><oc:id>4</oc:id><oc:display-name>work in progress</oc:display-name><oc:user-visible>true</oc:user-visible><oc:user-assignable>true</oc:user-assignable><oc:can-assign>true</oc:can-assign></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response></d:multistatus>
```

PROPFINDs on the documents itself:
 
PROPFIND https://cloud.nextcloud.com/remote.php/dav/systemtags-relations/files/755194

Reponse:

```
<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns"><d:response><d:href>/remote.php/dav/systemtags-relations/files/755194/</d:href><d:propstat><d:prop><oc:id/><oc:display-name/><oc:user-visible/><oc:user-assignable/><oc:can-assign/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response></d:multistatus>
```

I get the 404 in both cases: where the tag on a document works and where it doesn't.

**Comment by blizzz** [comment]
A POST should be fired when a tag was set, that's what happening for me.

Do you own that file or is it a shared one?

**Comment by mathiasconradt** [comment]
I'm not setting a tag (yet), I just click on "tag" word in order to open the tag input field. And that already does not always work, even before I submit any tag.
I can do a screenshare to show you.

**Comment by blizzz** [comment]
OK, tried it again, clicking on Tags to get into the text input works for the first file (more precise: for the first file of which I open the sidebar) after a page load, but not any other, and also not when returning to the first file. Seems the click action is assigned once,  but not any further.

**Comment by blizzz** [comment]
It's a backbone view and it seems that the click-event-binding get's lost.

**Comment by blizzz** [comment]
@MorrisJobke nope, the tag list does not even open for him on the second file

**Comment by blizzz** [comment]
fix in #10782

**Comment by SNAKE83** [comment]
Hi, any updates on solution to this problem?
I just installed NC14, few times, cause I could not understand what is happening, but it does not work.
Tag "link/button" works only once on page reload, after being used on one file, it does not open until page reload.

**Comment by blizzz** [comment]
@SNAKE83  #10782  is not merged yet, thus there isn't any backport to 14. → 14.0.2 is the next candidate.

**Comment by SNAKE83** [comment]
> @SNAKE83 #10782 is not merged yet, thus there isn't any backport to 14. → 14.0.2 is the next candidate.
:+1:  Thank You, so there is hope :)

**Comment by MorrisJobke** [comment]
Fix was reverted as it broke the sharing entries -> reopen

**Comment by Troublicious** [comment]
I can confirm the problem still exists in 14.0.2

**Comment by RuudschMaHinda** [comment]
Confirming this still existing in 14.0.3

**Comment by DBJRdev** [comment]
We encounter the same problem, appearing on two different NC instances (both NC 14.0.3, one updated since NC 12.x, the other started with NC 14), we tested with
* Firefox 62 on Windows 7
* Firefox 63 on Ubuntu 18.10
* Chrome 69 on both systems.

**Comment by RuudschMaHinda** [comment]
#12231 I _think_ this is related and _might_ fix it. Can't test it right now though.

**Comment by violoncelloch** [comment]
#12231 unfortunately doesn't fix this (tested with current master) but makes it worse (tags not working at all) but as this issue got recognized in #12320 we can hopefully expect a fix soon

**Comment by voxdemonix** [comment]
Bug present on 14.0.3 production too.

**Comment by SNAKE83** [comment]
How could it be released with a bug of something that was working earlier?
This is a very weird policy, makes You think what else is working not as supposed to be, maybe security?

**Comment by voxdemonix** [comment]
> How could it be released with a bug of something that was working earlier?

@SNAKE83 Evolve code without have any bug is a difficult art.

**Comment by MorrisJobke** [comment]
Most likely fixed by #12577. The backport will come to 14.0.5.

**Comment by SNAKE83** [comment]
Working in 15.0.0

**Comment by TueTchen** [comment]
Please Re-Open for 18.0.4 (just updated from 18.0.3).
In 18.0.3 it happened "sometimes" but often could be solved by refresh, sometimes by re-login.
But now I'm not able to Tag new files anymore.
Thank you for soon solutions in advance - yours, Peter alias TueTchen, Germany

**Comment by blizzz** [comment]
This is one and a half-year old. Please open a new issue with full details.
